src/app/code/community/AvS/FastSimpleImport/Model/Import/Entity/Product.php

```diff
@@ -1612,17 +1612,26 @@ protected function _getUploader()
     }
 
     /**
-     * @param $filePath
+     * @param $fileName
      * @return bool
      */
     protected function moveDownloadableFile($fileName)
     {
-        $filePath = $this->_getUploader()->getTmpDir() . $fileName;
-        $destDir = Mage::getConfig()->getOptions()->getMediaDir() . '/downloadable/files/links' . $fileName;
-
+        $filePath      = $this->_getUploader()->getTmpDir() . $fileName;
+        $basePath      = Mage::getModel('downloadable/link')->getBasePath();
+        $destDirectory = dirname(Mage::helper('downloadable/file')->getFilePath($basePath, $fileName));
+        // make sure that the destination directory exists!
+        $ioObject = new Varien_Io_File();
+        try {
+            $ioObject->open(array('path' => $destDirectory));
+        } catch (Exception $e) {
+            $ioObject->mkdir($destDirectory, 0777, true);
```

Comment:
First of all, thanks for your review! But I think we should keep 777 in order to stick to the Magento standard. Because they actually create the folder as 777 if you upload a file in a downloadable product. See https://github.com/OpenMage/magento-mirror/blob/505e8e982ad5f4fd22005060564cad210aab7b6f/app/code/core/Mage/Downloadable/Helper/File.php#L108. Do you agree based on this background?